spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/Spark3BinPackStrategy.java

```diff
@@ -67,12 +68,18 @@ public Table table() {
           .option(SparkReadOptions.FILE_OPEN_COST, "0")
           .load(table.name());
 
+      boolean requiresRepartition = !filesToRewrite.get(0).spec().equals(table.spec());
+
+      // Invoke a shuffle if the partition spec of the incoming partition does not match the table
+      String distributionMode = requiresRepartition ? DistributionMode.RANGE.modeName() :
```

Comment:
And also general question here for learning, what's the reason for range and not hash?

Reply:
Range is generally safer than hash when we don't know what the underlying distribution really is. For example if we have many small partitions a hash distribution would be a spark shuffle task per partition, with range it will attempt to sample and more evenly distribute the work within the spark shuffle (this also can take advantage of sort columns if they exist).